src/metrics/mod.rs

```diff
@@ -16,68 +19,197 @@ pub static PERCENTILES: &[(&str, f64)] = &[
     ("p9999", 99.99),
 ];
 
-pub struct Snapshot {
-    pub prev: HashMap<Metrics, u64>,
+pub struct HistogramsSnapshot {
+    pub previous: HashMap<String, metriken::histogram::Snapshot>,
+    pub deltas: HashMap<String, metriken::histogram::Snapshot>,
+}
```

Comment:
Nit: I think we should either move this down next to its implementation or move `CountersSnapshot` here so that all the structure declarations are together.

Reply:
Will fix.